libraries/pbrlib/genglsl/lib/mx_microfacet_diffuse.glsl

```diff
@@ -2,28 +2,87 @@
 
 // Based on the OSL implementation of Oren-Nayar diffuse, which is in turn
 // based on https://mimosa-pudica.net/improved-oren-nayar.html.
-float mx_oren_nayar_diffuse(vec3 L, vec3 V, vec3 N, float NdotL, float roughness)
+float mx_oren_nayar_diffuse(float NdotV, float NdotL, float LdotV, float roughness)
 {
-    float LdotV = clamp(dot(L, V), M_FLOAT_EPS, 1.0);
-    float NdotV = clamp(dot(N, V), M_FLOAT_EPS, 1.0);
     float s = LdotV - NdotL * NdotV;
     float stinv = (s > 0.0f) ? s / max(NdotL, NdotV) : 0.0;
 
-    float sigma2 = mx_square(roughness * M_PI);
+    float sigma2 = mx_square(roughness);
     float A = 1.0 - 0.5 * (sigma2 / (sigma2 + 0.33));
     float B = 0.45 * sigma2 / (sigma2 + 0.09);
 
```

Comment:
Note that this implementation is not actually the Fujii model as described in his blog post.

The correct form for that would be:
```glsl
    float stinv = (s > 0.0f) ? s / max(NdotL, NdotV) : s;
    float A = 1.0 / (1.0 + (0.5 - 2.0/(3.0*PI)) * sigma);
    float B = sigma * A;
    return A + B * stinv;
```
Otherwise this is just "qualitative Oren-Nayar", which has dark circle artifacts. (Both models suffer from fairly extreme darkening also).

And I think the `sigma` should just equal `roughness`, there's no need for the squaring, at least there's nothing in the blog post or other literature that suggest that should be done.

Reply:
For now, we're aligning the GLSL implementation of Oren-Nayar with the OSL implementation, which is defined here, and let me know if you notice any divergence between the two:

https://github.com/AcademySoftwareFoundation/OpenShadingLanguage/blob/7fe43cec6e947e584cecfda426d83a77df5af05a/src/testrender/shading.cpp#L479

To your second point, we're treating sigma and roughness as synonyms here, and we're only squaring the sigma value to generate sigma2 (the square of sigma).

Reply:
I've updated the comment to clarify that OSL is the source for the GLSL implementation, and have removed the reference to Fujii.